org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AspectjFilter.java

```diff
@@ -0,0 +1,206 @@
+ * This filter ignores these 3 method and the call.
+ * <p>
+ * Every class containing advised methods, where the advice takes a JoinPoint as parameter,
+ * will contain the {@code ajc$preClinit()} method which will be called from the static
+ * initializer ({@code <clinit>()}).
+ * This filter ignores this method, and its call.
+ * <p>
+ * If the static initializer ({@code <clinit>()}) only contains the calls to
+ * {@code ajc$preClinit()} and/or {@code ajc$postClinit()}, its completely ignored.
+ * <p>
+ * For every JoinPoint which is advised by an around advice, an {@code $AjcClosure}-class
+ * is generated (unless the around advice is inlined).
+ * This filter ignored all methods ins such classes.
+ */
+public class AspectjFilter implements IFilter {
+
+    private static final String AJ_SYNTHETIC_ATTRIBUTE = "org.aspectj.weaver.AjSynthetic";
+    private static final Pattern AJC_CLOSURE_PATTERN = Pattern.compile(".*\\$AjcClosure\\d+");
+
+    public void filter(MethodNode methodNode, IFilterContext context, IFilterOutput output) {
+
+        if (isAroundClosureClass(context)) {
+            output.ignore(methodNode.instructions.getFirst(), methodNode.instructions.getLast());
+            return;
+        }
+
+        if (isWellKnownSyntheticMethod(methodNode)) {
+            output.ignore(methodNode.instructions.getFirst(), methodNode.instructions.getLast());
+            return;
+        }
+
+        if (methodNode.name.equals("<clinit>")) {
+            checkStaticInitializer(methodNode, output);
+        }
+    }
+
+    private boolean isWellKnownSyntheticMethod(MethodNode methodNode) {
+        if (methodNode.name.equals("aspectOf") && isAjSynthetic(methodNode)) {
+            return true;
+        }
+        if (methodNode.name.equals("hasAspect") && isAjSynthetic(methodNode)) {
+            return true;
+        }
+        if (methodNode.name.equals("ajc$postClinit") && isAjSynthetic(methodNode)) {
+            return true;
+        }
+        if (methodNode.name.equals("ajc$preClinit") && ((methodNode.access & Opcodes.ACC_SYNTHETIC) != 0)) {
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isAroundClosureClass(IFilterContext context) {
+        return context.getSuperClassName().equals("org/aspectj/runtime/internal/AroundClosure")
+                && AJC_CLOSURE_PATTERN.matcher(context.getClassName()).matches();
+    }
+
+    /**
+     * This method finds calls to ajc$preClinit() and ajc$postClinit() and ignores them.
+     *
+     * @param methodNode The {@literal <clinit>()}-Method
+     * @param output
+     */
+    private void checkStaticInitializer(MethodNode methodNode, IFilterOutput output) {
+        MethodInsnNode preClinitNode = null;
+        MethodInsnNode postClinitNode = null;
+
+        for (AbstractInsnNode node = methodNode.instructions.getFirst();
+             node != null;
+             node = node.getNext()) {
+
+            if (node.getOpcode() != Opcodes.INVOKESTATIC) {
```

Comment:
Can you finally answer question asked during previous review ( https://github.com/jacoco/jacoco/pull/710#discussion_r202324700 ):
Why not use `AbstractMatcher`? which in particular has `skipNonOpcodes`

Reply:
The detection of calls to `ajc$preClinit` and `ajc$postClinit` now uses `AbstractMatcher` subclasses.